src/nodes.coffee

```diff
@@ -102,6 +136,54 @@ exports.Base = class Base
         parts.push    @makeCode ")"
     parts
 
+  compileCommentFragments: (o, node, fragments) ->
+    return fragments unless node.comments
+    # This is where comments, that are attached to nodes as a `comments`
+    # property, become `CodeFragment`s. “Inline block comments,” e.g.
+    # `/* */`-delimited comments that are interspersed within code on a line,
+    # are added to the current `fragments` stream. All other fragments are
+    # attached as properties to the nearest preceding or following fragment,
+    # to remain stowaways until they get properly output in `compileComments`
+    # later on.
+    unshiftCommentFragment = (commentFragment) ->
+      if commentFragment.unshift
+        # Find the first non-comment fragment and insert `commentFragment`
+        # before it.
+        unshiftAfterComments fragments, commentFragment
+      else
+        if fragments.length isnt 0
+          precedingFragment = fragments[fragments.length - 1]
+          if commentFragment.newLine and precedingFragment.code isnt '' and
+             not /\n\s*$/.test precedingFragment.code
+            commentFragment.code = "\n#{commentFragment.code}"
+        fragments.push commentFragment
+
+    for comment in node.comments when comment not in @compiledComments
+      @compiledComments.push comment # Don’t output this comment twice.
```

Comment:
Is this necessary when combined with the `ignoreTheseCommentsTemporarily` above? What case does it cover?

Also, since `compiledComments` is a prototype property it will be the same reference for all nodes, is that the intention?

```coffee
class Base
  foo: []
class A extends Base
class B extends Base
a = new A
a.foo.push 1
console.log new B().foo # [ 1 ]
```